pcdsdevices/lasers/rfof.py

```diff
@@ -13,17 +13,242 @@
+
+
+"""Instrumentation Technologies Low-Jitter Continuous Wave"""
+"""Reference Clock Transfer System"""
+
+
+class ItechRfofRx(BaseInterface, Device):
+
+    """Itech RFoF receiver PVs."""
+
+    rx_pc1 = Cpt(EpicsSignalRO, ':RX_PC1', kind='normal',
+                 doc='Rx internal lock (uV)')
+    rx_pd1_i = Cpt(EpicsSignalRO, ':RX_PD1_I', kind='normal',
+                   doc='Compenstated link photodiode current (uA)')
+    rx_pd2_i = Cpt(EpicsSignalRO, ':RX_PD2_I', kind='normal',
+                   doc='Low-noise link photodiode current (uA)')
+    rx_rfout = Cpt(EpicsSignalRO, ':RX_RFOUT', kind='normal',
+                   doc='Output RF power (dBm)')
+    rx_pd1_rf = Cpt(EpicsSignalRO, ':RX_PD1_RF', kind='normal',
+                    doc='Compensated link photodiode RF power (dBm)')
+    rx_pd2_rf = Cpt(EpicsSignalRO, ':RX_PD2_RF', kind='normal',
+                    doc='Low-noise link photodiode RF Power (dBm)')
+    rx_orf_s = Cpt(EpicsSignalRO, ':RX_ORF_S', kind='normal',
+                   doc='Output RF power setpoint (dBm)')
+    rx_att_v = Cpt(EpicsSignalRO, ':RX_ATT_V', kind='normal',
+                   doc='Variable attenuator voltage (V)')
+    rx_ps_v = Cpt(EpicsSignalRO, ':RX_PS_V', kind='normal',
+                  doc='Phase shifter voltage (V)')
+    rx_stp_t = Cpt(EpicsSignalRO, ':RX_STP_T', kind='normal',
+                   doc='Modules temperature setpoint (C)')
+    rx_t = Cpt(EpicsSignalRO, ':RX_T', kind='normal',
+               doc='Rx board temperature (C)')
+    rx_opt_t = Cpt(EpicsSignalRO, ':RX_OPT_T', kind='normal',
+                   doc='Optical assembly temperature (C)')
+    rx_spl_t = Cpt(EpicsSignalRO, ':RX_SPL_T', kind='normal',
+                   doc='Rx spool temperature (C)')
+    rx_ext_h = Cpt(EpicsSignalRO, ':RX_EXT_H', kind='normal',
+                   doc='External relative humidity (%)')
+    rx_exth_t = Cpt(EpicsSignalRO, ':RX_EXTH_T', kind='normal',
+                    doc='External temperature (humidity sensor) (C)')
+    rx_ext_p = Cpt(EpicsSignalRO, ':RX_EXT_P', kind='normal',
+                   doc='External pressure (mbar)')
+    rx_extp_t = Cpt(EpicsSignalRO, ':RX_EXTP_T', kind='normal',
+                    doc='External temperature (pressure sensor) (C)')
+    rx_int_h = Cpt(EpicsSignalRO, ':RX_INT_H', kind='normal',
+                   doc='Internal relative humidity (%)')
+    rx_inth_t = Cpt(EpicsSignalRO, ':RX_INTH_T', kind='normal',
+                    doc='Internal temperature (humidity sensor) (C)')
+    rx_int_p = Cpt(EpicsSignalRO, ':RX_INT_P', kind='normal',
+                   doc='Internal pressure (mbar)')
+    rx_intp_t = Cpt(EpicsSignalRO, ':RX_INTP_T', kind='normal',
+                    doc='Internal temperature (pressure sensor) (C)')
+
+
+class ItechRfofTx(BaseInterface, Device):
+
+    """Itech RFoF transmitter PVs."""
+
+    # Transmitter adds a single control PV
+    tx_pc1 = Cpt(EpicsSignalRO, ':TX_PC1', kind='normal',
```

Comment:
All of these have an identical prefix `tx_`. You should consider removing this prefix entirely.

It'd also be a good idea to give them names that are sensible for the end-user. You did a great job including the documentation string there. It could be a good hint as to what the attribute names should be.

For example: 
"pc1" means absolutely nothing to me. However, "low-drift optical link loop lock in uV" is very clear. This could mean that instead of `tx_pc1 = Cpt(...)` this could rather be `link_loop_lock = Cpt(...)`

Reply:
Is there a limit on the length of the name of the components? Some of these would be quite long...

Reply:
There is no specific limit to component name length. We do, however, keep our code lines to be < 80 characters in general.

Sensible component attribute names are an important thing (in my opinion), and coming up with appropriate ones does take a bit of practice.

I'd suggest formatting these differently if you start increasing the length or amount of information passed to the `Cpt`:
```python
class MyDevice(Device):
    long_component_name = Cpt(
        ComponentClass,
        "Prefix",
        kind="normal",
        doc="This is an explanation about what the component does",
    )
```

I think most would agree that's much more readable than combining it into two lines.

Reply:
But for something like the" Variable RF attenuator controller setpoint voltage", there isn't really a way to have that be short, but also descriptive. I could do something like var_rf_att_ctrl_sp_v, I guess?

Reply:
`var_rf_att_ctrl_sp_v` has almost lost all meaning with introducing a bunch of new acronyms unnecessarily. A good choice for a name really depends on the device - which I don't know enough about here to make the right suggestion for you.

It does not need to have _everything_ from the `doc` help string, just the important bits to make it uniquely identifiable and useful to the end-user.

Something like `attenuator_setpoint_voltage` or `rf_setpoint_voltage` could be sufficient, if it uniquely identifies its purpose among all of the other components.